packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -678,18 +680,22 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
 
   @override
   void didUpdateConfig(Navigator oldConfig) {
-    if (oldConfig.observer != config.observer) {
```

Comment:
I'd keep this equality check.  That lets a client who has a really complex set of observers avoid the loops if they do the work of re-using the same list when it doesn't change.